### DISCUSSION TITLE
Building entityrecognizer from json takes substantially shorter than loading model

### DISCUSSION THREAD

**Discussion by SjoerdBraaksma** [explosion/spaCy#12091] [Help: Best practices]
Hello!

In my current project I'm working with two custom entityrecognizers (one for names, one for places), with a large number of patters (500k).

I noticed that adding these pipes to the standard dutch model like so:

```
nlp = sp.load("nl_core_news_lg")

# Make config file
config_names = {
   "validate": False,
   "overwrite_ents": True,
   "ent_id_sep": "||",
}

config_places = {
   "validate": False,
   "overwrite_ents": False,
   "ent_id_sep": "||",
}

# Create EntityRuler
ruler_names= nlp.add_pipe("entity_ruler", "names_ruler", before="ner", config=config_names)
ruler_places= nlp.add_pipe("entity_ruler", "places_ruler", after="names_ruler", config=config_places)

# add respective patterns
with nlp.select_pipes(enable="ner"):
    ruler_names.add_patterns(names["namen"])
    ruler_places.add_patterns(names["plaatsen"])
```
This takes around 40 seconds. 

However, if I save this model and load it from disk (nlp.to_disk and spacy.load) takes about 4,5 minutes. Not much difference if I exclude all but the 2 rulers and the ner component.
Am I doing something wrong with saving and loading, or am I not implementing something that could give me a time save boost?

Thanks in advance for the help!

**Comment by shadeMe** [comment]
Thanks for the report and the example code. The slowdown is due to the fact that adding new patterns to the entity ruler component requires executing all the pipes preceding it to generate that internal `Doc` instances that facilitate the matching. 

We [previously considered](https://github.com/explosion/spaCy/pull/10944) exposing a `disable_pipes` parameter to workaround this behaviour but ultimately decided against it. So, if you have a performance-sensitive use case, we would suggest that you use the code above to minimize the overhead.

**Reply by adrianeboyd** [reply]
In addition, if your entity ruler patterns don't rely on attributes like `TAG`/`DEP`/`LEMMA` from the other pipeline components, you can move the `ner` and `entity_ruler` earlier in the pipeline to improve the loading time for this case, since the entity ruler runs all components up to its own position in the pipeline.

One option would be to reorder components in the pipeline when loading like this:

```python
nlp = spacy.load("nl_core_news_lg", config={"nlp": {"pipeline": ["ner", "tok2vec", "morphologizer", "tagger", "parser", "lemmatizer", "senter", "attribute_ruler"]}})
nlp.add_pipe("entity_ruler", after="ner")
```

We do have to know in advance for this pipeline that the `ner` component is an independent component that doesn't depend on `tok2vec` or other prior components.
